hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -698,7 +686,59 @@ public String getPayloadClass() {
    * Read the payload class for HoodieRecords from the table properties.
    */
   public String getRecordMergerStrategy() {
-    return getStringOrDefault(RECORD_MERGER_STRATEGY);
+    return getString(RECORD_MERGER_STRATEGY);
+  }
+
+  public static Triple<RecordMergeMode, String, String> inferCorrectMergingBehavior(RecordMergeMode recordMergeMode, String payloadClassName, String recordMergerStrategy) {
+    RecordMergeMode inferRecordMergeMode;
+    String inferPayloadClassName;
+    String inferRecordMergerStrategy;
+
+    if (isNullOrEmpty(payloadClassName)) {
+      if (isNullOrEmpty(recordMergerStrategy)) {
+        checkArgument(recordMergeMode != RecordMergeMode.CUSTOM, "Custom merge mode should only be used if you set a merge strategy");
+        if (recordMergeMode == null) {
+          inferRecordMergeMode = RECORD_MERGE_MODE.defaultValue();
+        } else {
+          inferRecordMergeMode = recordMergeMode;
+        }
+        inferRecordMergerStrategy = HoodieRecordMerger.getAvroMergerStrategyFromMergeMode(inferRecordMergeMode);
+      } else {
+        inferRecordMergerStrategy = recordMergerStrategy;
+        if (recordMergerStrategy.equals(DEFAULT_MERGER_STRATEGY_UUID)) {
+          inferRecordMergeMode = EVENT_TIME_ORDERING;
+        } else if (recordMergerStrategy.equals(OVERWRITE_MERGER_STRATEGY_UUID)) {
+          inferRecordMergeMode = OVERWRITE_WITH_LATEST;
+        } else {
+          checkArgument(!recordMergerStrategy.equals(PAYLOAD_BASED_MERGER_STRATEGY_UUID), "Payload based strategy should only be used if you have a custom payload");
+          inferRecordMergeMode = CUSTOM;
+        }
+      }
+      inferPayloadClassName = HoodieRecordPayload.getAvroPayloadForMergeMode(inferRecordMergeMode);
+    } else {
+      inferPayloadClassName = payloadClassName;
+      if (payloadClassName.equals(DefaultHoodieRecordPayload.class.getName())) {
+        //TODO: after mergers are safe to use on the write side, add this check and get rid of the if
+        // checkArgument(isNullOrEmpty(recordMergerStrategy) || recordMergerStrategy.equals(DEFAULT_MERGER_STRATEGY_UUID), "Record merge strategy cannot be set if a merge payload is used");
+        if (isNullOrEmpty(recordMergerStrategy) || recordMergerStrategy.equals(DEFAULT_MERGER_STRATEGY_UUID)) {
+          inferRecordMergeMode = EVENT_TIME_ORDERING;
+          inferRecordMergerStrategy = DEFAULT_MERGER_STRATEGY_UUID;
+        } else {
+          inferRecordMergeMode = CUSTOM;
+          inferRecordMergerStrategy = recordMergerStrategy;
+        }
+      } else if (payloadClassName.equals(OverwriteWithLatestAvroPayload.class.getName())) {
+        checkArgument(isNullOrEmpty(recordMergerStrategy) || recordMergerStrategy.equals(OVERWRITE_MERGER_STRATEGY_UUID), "Record merge strategy cannot be set if a merge payload is used");
+        inferRecordMergeMode = OVERWRITE_WITH_LATEST;
+        inferRecordMergerStrategy = OVERWRITE_MERGER_STRATEGY_UUID;
```

Comment:
For merge modes of `OVERWRITE_WITH_LATEST` and `EVENT_TIME_ORDERING`, should we avoid persisting the record merge strategy ID?